docs/design/node_agent_config-design.md

```diff
@@ -0,0 +1,517 @@
+					nullable: true
+					description: 'Requests describes the minimum amount of compute resources required. If Requests is omitted for a container, it defaults to Limits if that is explicitly specified, otherwise to an implementation-defined value. More info: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/'
+					type: object
+					nullable: true
+				type: object
+			tolerations:
+				description: tolerations defines the list of tolerations to be applied to Restic daemonset
+				items:
+				description: The pod this Toleration is attached to tolerates any taint that matches the triple <key,value,effect> using the matching operator <operator>.
+				properties:
+					effect:
+					description: Effect indicates the taint effect to match. Empty means match all taint effects. When specified, allowed values are NoSchedule, PreferNoSchedule and NoExecute.
+					type: string
+					key:
+					description: Key is the taint key that the toleration applies to. Empty means match all taint keys. If the key is empty, operator must be Exists; this combination means to match all values and all keys.
+					type: string
+					operator:
+					description: Operator represents a key's relationship to the value. Valid operators are Exists and Equal. Defaults to Equal. Exists is equivalent to wildcard for value, so that a pod can tolerate all taints of a particular category.
+					type: string
+					tolerationSeconds:
+					description: TolerationSeconds represents the period of time the toleration (which must be of effect NoExecute, otherwise this field is ignored) tolerates the taint. By default, it is not set, which means tolerate the taint forever (do not evict). Zero and negative values will be treated as 0 (evict immediately) by the system.
+					format: int64
+					type: integer
+					value:
+					description: Value is the taint value the toleration matches to. If the operator is Exists, the value should be empty, otherwise just a regular string.
+					type: string
+				type: object
+				type: array
+			type: object
+		supplementalGroups:
+			description: supplementalGroups defines the linux groups to be applied to the Restic Pod
+			items:
+			format: int64
+			type: integer
+			type: array
+		timeout:
+			description: timeout defines the Restic timeout, default value is 1h
+			type: string
+		type: object
+```
+
+### Validations
+
+It is important to warn user that both options `restic` and `nodeagentconfig` are not allowed together.
+
+### Deprecation of the `restic` coiguration option from OADP 1.4+
+
+Since the `restic` configuration option will be deprecated with the OADP 1.4, it is important to inform user to prepare for such deprecation. There were few alternatices considered (see [Alternatives Considered](#alternatives-considered)). We will have three places where the deprecation information will be presented to the user:
+1. Description of the `resic` property will have the deprecation warning
```

Comment:
@sseago Could you point me to example how this could work? I checked and we can add warning of deprecation of entire CRD, but really one field. Please see also the considered alternatives for the deprecation methods, which I've discussed with @shawn-hurley, however I may be not aware of other options to inform user of the deprecation.

Reply:
@mpryc I suspect that this needs to be handled as part of validation logic in the oadp controller rather than a controller-runtime flag. Basically when validating, if this field is present we need to add a warning condition. @shubham-pampattiwar  are we doing anything similar elsewhere? Adding DPA warning conditions during validation?

Reply:
I think that we can do in order of what I would prefer:

1. Validation webhook that can add a warning IF it is set. 
2. Create a condition, about deprecation like Scott mentioned.
3. I also think an event on the object could work as well, if we don't want to do the above. 

I do love this discussion though, forcing the user to be aware of this is really good!

Reply:
See below, but that's open to discuss:

```
1. Description of the `resic` property will have the deprecation warning
2. Application log will have `warning` message to inform user that the `restic` will be deprecated if the `restic` was used
3. Release notes for OADP 1.3 will contain information about new configuration option `nodeAgentConfig` that should be used instead of `restic`
```

Reply:
> I think that we can do in order of what I would prefer:
> 
> 1. Validation webhook that can add a warning IF it is set.
Wasn't there a general problem with webhooks that were problematic and in some events could cause more trouble? I am all for the best technical way to handle such deprecation warning, just not experience enough to make the call which way to go.

> 3. I also think an event on the object could work as well, if we don't want to do the above.
I am not sure if that's technically correct for one particular field within API version ? The doc is about deprecation of entire API and replacing it with the newer version of API. We are not going to bump the API version, right?

https://kubernetes.io/blog/2020/09/03/warnings/#deprecation-warnings

Reply:
I think we have some leeway here as we are in v1alpha1.

If we wanted to, an option here is to create a conversion webhook and bump to, say, v1alpha2 or v1beta1. 

I would just as soon just tell the user from one of the mechanisms that the field no longer has meaning in the future.

Reply:
Changed to have an event within the DPA. Please note there is also extended **Deprecation warnings** with explanation of other options considered. It's also visible on the `oc describe dpa....` under `Events`

![Screenshot from 2023-08-02 14-16-36](https://github.com/openshift/oadp-operator/assets/16115240/57eb6f0f-f7c8-42a0-8d2a-ab1e204547ba)